lib/schemas.js

```diff
@@ -129,19 +136,37 @@ function getSchemaAnyway (schema, jsonShorthand) {
  *
  * @param {object} context the request context
  * @param {number} statusCode the http status code
+ * @param {string} contentType the reply content type
  * @returns {function|boolean} the right JSON Schema function to serialize
  * the reply or false if it is not set
  */
-function getSchemaSerializer (context, statusCode) {
+function getSchemaSerializer (context, statusCode, contentType) {
   const responseSchemaDef = context[kSchemaResponse]
   if (!responseSchemaDef) {
     return false
   }
   if (responseSchemaDef[statusCode]) {
+    if (responseSchemaDef[statusCode].constructor === Object) {
+      const type = contentType.split(';')[0]
```

Comment:
We should add a guard here if content type is missing

Reply:
Can it be missing? I've looked after this for a while to write a test and almost found that it's always set if not specified, but I'm not very sure though, wdyt?
https://github.com/fastify/fastify/blob/aa43e2de2cac18d7654c6a88e5437f8b0854872e/lib/reply.js#L133-L169